File: kotlinsensors/src/main/java/com/lexicalninja/kotlinsensors/serializers/CyclingMeasurementData.java
package com.lexicalninja.kotlinsensors.serializers;

import androidx.annotation.Nullable;

/**
 * Created by Saxton on 7/21/16.
 */
public interface CyclingMeasurementData {
    public double getTimeStamp();

    @Nullable
    public Integer getCumulativeCrankRevolutions();

    @Nullable
    public Integer getCumulativeWheelRevolutions();

    @Nullable
    public Integer getLastCrankEventTime();

    @Nullable
    public Short getLastWheelEventTime();
}

File: kotlinsensors/src/main/java/com/lexicalninja/kotlinsensors/serializers/CyclingMeasurementData.kt
package com.lexicalninja.kotlinsensors.serializers

/**
 * Created by Saxton on 7/21/16.
 */
interface CyclingMeasurementData {
    var timestamp: Double
    var cumulativeCrankRevolutions: Int?
    var cumulativeWheelRevolutions: Int?
    var lastCrankEventTime: Int?
    var lastWheelEventTime: Short?
}